emperor/support_files/js/controller.js

```diff
@@ -262,12 +263,54 @@ define([
       selector: '#' + scope.$divId.attr('id') + ' .emperor-plot-wrapper',
       trigger: 'none',
       items: {
+        'saveState': {
+          name: 'Save current settings',
+          icon: 'edit',
+          callback: function(key, opts) {
+            scope.saveConfig();
+          }
+        },
+        'loadState': {
+          name: 'Load saved settings',
+          icon: 'paste',
+          callback: function(key, opts) {
+            if (!FileReader) {
+              alert("Your browser does not support file loading. Please upgrade your browser.");
+              return;
+            }
+            var file = $('<input type="file">');
+            file.on('change', function(evt) {
+              var f = evt.target.files[0];
+              // With help from
+              // http://www.htmlgoodies.com/beyond/javascript/read-text-files-using-the-javascript-filereader.html
+              var r = new FileReader();
+              r.onload = function(e) {
+                try {
+                  var json = JSON.parse(e.target.result);
+                } catch (e) {
```

Comment:
This `e` is shadowing the value of the argument `e` in the `r.onload` callback, can you change this to be either `error` or something else?